pages/api/inventory/index.js

```diff
@@ -7,13 +7,23 @@ import { DEFAULT_ROWS_LIMIT } from "@/utils/api.util";
 const getAllInventory = async (req, res) => {
   console.log("Get all inventory Request Start");
 
-  const { limit, offset, attributes = [] } = req.query;
+  const { limit, offset, attributes = [], filters } = req.query;
   const options = {};
   options.limit = limit ? limit : DEFAULT_ROWS_LIMIT;
   options.offset = offset ? offset : 0;
   if (attributes.length) {
     options.attributes = JSON.parse(attributes);
   }
+  if (filters) {
+    try {
+      const parsedFilters = JSON.parse(filters);
+      if (parsedFilters.length > 0) {
+        options.filters = parsedFilters;
+      }
+    } catch (error) {
+      console.log("Error parsing filters:", error);
+    }
```

Comment:
Make a util method for checking if a value can be parsed

```suggestion
    /**
 * Utility function to check if a value is a valid JSON string and parse it.
 * @param {string} value - The string to parse.
 * @returns {Object|Array|null} - The parsed JSON object/array if valid, or null if parsing fails.
 */
const isParsableJSON = (value) => {
  if (typeof value !== 'string') {
    return null; // Not a string, can't parse
  }

  try {
    const parsed = JSON.parse(value);
    return parsed; // Return the parsed object/array
  } catch (error) {
    console.error("Failed to parse JSON:", { error, value });
    return null; // Return null in case of failure
  }
};
```
```suggestion
const parsedFilters = isParsableJSON(filters);

if (parsedFilters && parsedFilters.length > 0) {
  options.filters = parsedFilters;
}
```